boa/src/syntax/lexer/regex.rs

```diff
@@ -0,0 +1,109 @@
+use super::{Cursor, Error, Tokenizer};
+use crate::syntax::ast::Position;
+use crate::syntax::lexer::Token;
+use std::io::{self, ErrorKind, Read};
+
+/// Regex literal lexing.
+///
+/// Lexes Division, Assigndiv or Regex literal.
+///
+/// Expects: Initial '/' to already be consumed by cursor.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///  - [MDN documentation][mdn]
+///
+/// [spec]: https://www.ecma-international.org/ecma-262/#sec-literals-regular-expression-literals
+/// [mdn]:
+#[derive(Debug, Clone, Copy)]
+pub(super) struct RegexLiteral;
+
+impl RegexLiteral {
+    /// Creates a new regex literal lexer.
+    pub(super) fn new() -> Self {
+        Self {}
+    }
+}
```

Comment:
I think we don't need this.

Reply:
Yeah, it can be shortened as I don't need to store anything in the struct.